### PR TITLE
fix: add compatibility for incorrect file object input

### PR DESCRIPTION
**Pull request by leermao**
### Checklist

- [x] This change requires documentation updates, and I have updated the relevant documentation. If the documentation has not been updated, please create a documentation update issue and link it here
- [x] The changes are already covered by tests, and I have adjusted the test coverage for the changed parts
- [x] The newly added code logic is also covered by tests
- [x] This change adds dependencies, and they are placed in dependencies and devDependencies
- [x] This change includes adding or updating npm dependencies, and it does not result in multiple versions of the same dependency [check the diff of pnpm-lock.yaml]

### Summary by AIGNE

Release Notes:

- Bug Fix: Enhanced file handling validation to prevent processing of invalid or corrupted files
- Test: Added comprehensive test coverage for file processing edge cases
- Refactor: Improved type checking and error handling for file uploads

These changes strengthen the reliability of file processing functionality, reducing the likelihood of errors when handling various file types and formats. Users will experience more robust file upload capabilities with better error messages for invalid inputs.

### CODE REVIEW

File: observability/api/api/server/utils/save-files.ts [hunk 1]
```diff
@@ -29,7 +29,13 @@ const saveFiles = async (
         mkdirSync(options.dataDir, { recursive: true });
       }
 
-      if ("type" in file && file.type === "file" && typeof file.data === "string") {
+      if (
+        typeof file === "object" &&
+        file !== null &&
```
Comment (li-yechao) [inline observability/api/api/server/utils/save-files.ts]: `file !== null` -> `file && typeof file === 'object'`

Comment (li-yechao) [reply observability/api/api/server/utils/save-files.ts]: 另外写文件时加上 try catch，如果失败则跳过文件保存，但需要输出 warn 日志
还有可以在检查中加入判断 data 长度，超过 200 个字符才处理，否则跳过
